### PR TITLE
Update version 1.1.3 -> 1.1.4

### PR DESCRIPTION
**Pull request by finnif**

### CODE REVIEW

File: Assets/ConnectApp/Components/EggButton.cs [hunk 1]
```diff
@@ -59,30 +67,45 @@ class _EggButtonState : State<EggButton>, TickerProvider {
         }
 
         public override Widget build(BuildContext context) {
+            var egg = Image.asset(
+                "image/egg",
+                width: 21,
+                height: 21
+            );
+
+            if (this.widget.isNationalDay) {
```
Comment (zhuxingwei) [inline Assets/ConnectApp/Components/EggButton.cs]: how about  => egg = this.widget.isNationalDay ? XX : YY;

File: Assets/ConnectApp/Utils/CImageUtils.cs [hunk 2]
```diff
@@ -40,45 +40,64 @@ public static class CImageUtils {
...
-                        padding: padding,
-                        child: Image.asset(
-                            "image/official-badge",
-                            height: 18,
-                            width: 18
-                        )
-                    );
-                }
-            }
+            Widget w1 = new Container();
```
Comment (zhuxingwei) [inline Assets/ConnectApp/Utils/CImageUtils.cs]: can we refine these lines a bit:

The outcomes is just a List<Widget>, why not add widget to this list and replace hasFirst => widgetList.isEmpty() ?

the constructor of w1 may be moved into a new function, e.g.,  CreateRoleBadgeIcon() which returns w1 (null indicates nothing). Then we can have a lot of CreateXXXIcon() when more badges are introduced?

File: Assets/ConnectApp/Utils/HttpManager.cs [hunk 3]
```diff
@@ -205,9 +205,34 @@ public static class HttpManager {
                 StoreProvider.store.dispatcher.dispatch(new ScanEnabledAction {
                     scanEnabled = scan
                 });
+
+                DateTime endTime;
```
Comment (zhuxingwei) [inline Assets/ConnectApp/Utils/HttpManager.cs]: use very similar codes in both the Try and Catch section, can we make it a function and call ?

File: Assets/ConnectApp/Utils/CImageUtils.cs (hunk 2 above)
Comment (finnif) [reply Assets/ConnectApp/Utils/CImageUtils.cs]:
```
public static Widget GenBadgeImage(List<string> badges, string license, EdgeInsets padding) {
            
            var badgeList  = new List<Widget>();
            Widget badgeWidget = null;
            
            if (license.isNotEmpty()) {
                if (license == "UnityPro") {
                    badgeWidget = Image.asset(
                        "image/pro-badge",
                        height: 15,
                        width: 26
                    );
                }

                if (license == "UnityPersonalPlus") {
                    badgeWidget = Image.asset(
                        "image/plus-badge",
                        height: 15,
                        width: 30
                    );
                }
            }

            if (badges != null && badges.isNotEmpty()) {
                if (badges.Any(badge => badge.isNotEmpty() && badge.Equals("official"))) {
                    badgeWidget = Image.asset(
                        "image/official-badge",
                        height: 18,
                        width: 18
                    );
                }
            }

            if (badgeWidget != null) {
                badgeList.Add(item: badgeWidget);
            }

            if (isNationalDay) {
                if (badgeList.Count >= 1) {
                    badgeList.Add(new SizedBox(width: 4));
                }
                badgeList.Add(Image.asset(
                    "image/china-flag-badge",
                    height: 14,
                    width: 16
                ));
            }
            
            if (badgeList.Count > 0) {
                return new Container(
                    padding: padding,
                    child: new Row(
                        children: badgeList
                    )
                );
            }
            
            return new Container();
        }
```

**Pull request merged**